lib/cn/cn_tree.c

```diff
@@ -1965,20 +1961,263 @@ static merr_t
 cn_subspill_apply(struct subspill *ss)
 {
     struct cn_compaction_work *w = ss->ss_work;
-    merr_t err;
+    merr_t err = 0;
+    bool delete_node = false;
+    struct cn_tree_node *outnode;
+
+    if (!ss->ss_added)
+        return 0;
+
+    if (ss->ss_is_zspill) {
+        err = cn_move(w, w->cw_node, ss->ss_zspill.zsp_src_list,
+                      w->cw_kvset_cnt, delete_node, ss->ss_node);
+
+        w->cw_output_nodev = &outnode;
+
+    } else {
+        err = cn_subspill_commit(ss);
+    }
 
-    err = cn_subspill_commit(ss);
     if (err)
         return err;
 
-    if (ss->ss_added) {
-        w->cw_output_nodev[0] = ss->ss_node;
-        w->cw_checkpoint(w);
-    }
+    w->cw_output_nodev[0] = ss->ss_node;
+    w->cw_checkpoint(w);
 
     return 0;
 }
 
+/* Tree must be locked when this is called */
+struct cn_tree_node *
+cn_kvset_can_zspill(struct kvset *ks, struct route_map *map)
+{
+    unsigned char maxkbuf[HSE_KVS_KEY_LEN_MAX];
+    const void *maxkey, *maxptkey, *minkey;
+    uint16_t maxklen, maxptklen, minklen;
+    struct route_node *minnode;
+
+    maxklen = minklen = maxptklen = 0;
+    maxptkey = NULL;
+
+    kvset_minkey(ks, &minkey, &minklen);
+    kvset_maxkey(ks, &maxkey, &maxklen);
+    kvset_max_ptkey(ks, &maxptkey, &maxptklen);
```

Comment:
Can you create function comments for `kvset_minkey` and `kvset_maxkey` to indicate they return the min/max of all keys including the prefix tombstones?